src/utils/hook.ts

```diff
@@ -375,6 +375,30 @@ export function useChartQueryWithCache<T>(
   })
 }
 
+export const useAnimationFrame = (callback: () => void, running: boolean = true) => {
+  const savedCallback = useRef(callback)
+  const requestId = useRef(0)
+
+  useEffect(() => {
+    savedCallback.current = callback
```

Comment:
It doesn't look like `useEffect` is needed here.
Code like `savedCallback` can be abstracted to https://ahooks.js.org/hooks/use-memoized-fn for use.